circuit/types/scalar/src/compare.rs

```diff
@@ -29,7 +30,7 @@ impl<E: Environment> Compare<Scalar<E>> for Scalar<E> {
         else {
             // If all scalar field elements are less than (MODULUS - 1)/2 on the base field,
             // we can perform an optimized check for `is_less_than` by casting the scalars onto the base field.
-            debug_assert!(E::ScalarField::modulus() < E::BaseField::modulus_minus_one_div_two());
+            debug_assert!(E::ScalarField::modulus() <= E::BaseField::modulus_minus_one_div_two());
```

Comment:
We should be able to leave this as is, no?

* `r = E::ScalarField::modulus() = 2111115437357092606062206234695386632838870926408408195193685246394721360383`, 
* `q = E::BaseField::modulus() = 8444461749428370424248824938781546531375899335154063827935233455917409239041`

Plugging these values into python tells me that that `(q - 1)/2 > r`

Reply:
Updated check.